languagetool-core/src/main/java/org/languagetool/language/identifier/LanguageIdentifier.java

```diff
@@ -0,0 +1,118 @@
+package org.languagetool.language.identifier;
+
+import com.optimaize.langdetect.text.TextFilter;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.Nullable;
+import org.languagetool.DetectedLanguage;
+import org.languagetool.Language;
+import org.languagetool.Languages;
+import org.languagetool.language.identifier.detector.CommonWordsDetector;
+import org.languagetool.language.identifier.detector.UnicodeBasedDetector;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public abstract class LanguageIdentifier {
+    private static final Pattern URL_REGEX = Pattern.compile("https?://[-_.?&~;+=/#%0-9A-Za-z]+");   // '%' has been added
```

Comment:
`protected` fields usually come before `private` ones, don't they?

Reply:
We got an `illegal forward reference` if we move the protected fields before the private ones.